spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/TaskServiceUtils.java

```diff
@@ -59,12 +60,28 @@ public static boolean isComposedTaskDefinition(String dsl) {
 	 */
 	public static String createComposedTaskDefinition(String graph,
 			TaskConfigurationProperties taskConfigurationProperties) {
+			return createComposedTaskDefinition(null, graph, taskConfigurationProperties);
+	}
+
+	/**
+	 * Creates a properly formatted CTR definition based on the graph provided.
+	 * @param graph the graph for the CTR to execute.
+	 * @param taskConfigurationProperties the properties that contain the name
+	 * of the CTR app to be launched.
+	 * @return String containing the CTR task definition.
+	 */
+	public static String createComposedTaskDefinition(String userComposedTaskRunnerName, String graph,
+			TaskConfigurationProperties taskConfigurationProperties) {
 		Assert.hasText(graph, "graph must not be empty or null");
 		Assert.notNull(taskConfigurationProperties,
 				"taskConfigurationProperties must not be null");
 		Assert.hasText(taskConfigurationProperties.getComposedTaskRunnerName(),
 				"taskConfigurationProperties.composedTaskRunnerName must not be null");
-		return String.format("%s --graph=\"%s\"", taskConfigurationProperties.getComposedTaskRunnerName(), graph);
+		String composedTaskRunnerName = taskConfigurationProperties.getComposedTaskRunnerName();
+		if(StringUtils.hasText(userComposedTaskRunnerName)) {
+			composedTaskRunnerName = userComposedTaskRunnerName;
```

Comment:
Since the custom composed task runner name should be registered, we can check early to see if it has in fact been registered to avoid discovering at runtime that it doesn't exist, for example due to a type or other user error.  Not sure this is the best spot to do that, I guess we can d such a check much earlier in the request.

Reply:
Updated based on your request.  It now checks to see if the specified ctr app is valid before this method is called.